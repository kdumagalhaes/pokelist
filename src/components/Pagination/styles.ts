import styled from 'styled-components'

export const Container = styled.div`
  display: flex;
  gap: 15px;

  .pagination-btn {
    display: flex;
    align-items: center;
    gap: 8px;
    padding: 1rem 1.2rem;
    background-color: ${({ theme }) => theme.blue};
    border-radius: 5px;
    border: none;
    color: ${({ theme }) => theme.yellow};
    font-weight: 700;
    transition: background-color 0.3s;

    &:hover {
      background-color: ${({ theme }) => theme.darkestBlue};
    }
  }

  .pagination-icon {
    width: 18px;
    height: 18px;
  }
`
