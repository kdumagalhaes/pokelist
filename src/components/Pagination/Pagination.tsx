import { Container } from './styles'
import LeftArrowIcon from '../../assets/icons/circle-left.svg'
import RightArrowIcon from '../../assets/icons/circle-right.svg'
import { usePokemon } from '../../context/PokeContext'

export function Pagination() {
  const { goToNextPage, goToPrevPage, prevPageUrl } = usePokemon()
  return (
    <Container>
      {/* if the user is on the first page, hide the previous button */}
      {prevPageUrl !== null ? (
        <button onClick={goToPrevPage} className="pagination-btn">
          <img
            className="pagination-icon"
            src={LeftArrowIcon}
            alt="left arrow"
          />
          Previous
        </button>
      ) : (
        ''
      )}
      <button onClick={goToNextPage} className="pagination-btn">
        <img
          className="pagination-icon"
          src={RightArrowIcon}
          alt="right arrow"
        />
        Next
      </button>
    </Container>
  )
}
