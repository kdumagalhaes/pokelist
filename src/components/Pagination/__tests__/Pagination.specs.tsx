import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Pagination } from '../Pagination'

describe('Pagination', () => {
  it('initial component should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Pagination />
      </BrowserRouter>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render previous button', () => {
    render(
      <BrowserRouter>
        <Pagination />
      </BrowserRouter>,
    )
    const previousButton = screen.getByRole('button', { name: /previous/i })
    expect(previousButton).toBeVisible()
  })

  it('should render next button', () => {
    render(
      <BrowserRouter>
        <Pagination />
      </BrowserRouter>,
    )
    const previousButton = screen.getByRole('button', { name: /next/i })
    expect(previousButton).toBeVisible()
  })
})
