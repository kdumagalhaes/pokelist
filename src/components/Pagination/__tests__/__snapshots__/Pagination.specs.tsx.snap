// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination initial component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe cvRPSc"
  >
    <button
      class="pagination-btn"
    >
      <img
        alt="left arrow"
        class="pagination-icon"
        src="circle-left.svg"
      />
      Previous
    </button>
    <button
      class="pagination-btn"
    >
      <img
        alt="right arrow"
        class="pagination-icon"
        src="circle-right.svg"
      />
      Next
    </button>
  </div>
</DocumentFragment>
`;
